percona-server-mongodb.36/ps-entry.sh

```diff
@@ -346,6 +343,10 @@ if [ "$originalArgOne" = 'mongod' ]; then
 		set -- "${mongodHackedArgs[@]}"
 	fi
 
+	sslMode="$(_mongod_hack_have_arg '--sslPEMKeyFile' "$@" && echo 'preferSSL' || echo 'disabled')" # "BadValue: need sslPEMKeyFile when SSL is enabled" vs "BadValue: need to enable SSL via the sslMode flag when using SSL configuration parameters"
+	_mongod_hack_ensure_arg_val --sslMode "$sslMode" "$@"
+	set -- "${mongodHackedArgs[@]}"
```

Comment:
hm, is it really needed to assign `mongodHackedArgs` to `$@` ?

Reply:
unfortunately we need that line due to 

```
_mongod_hack_ensure_arg_val() {
	local ensureArg="$1"; shift
	local ensureVal="$1"; shift
	_mongod_hack_ensure_no_arg_val "$ensureArg" "$@"
	mongodHackedArgs+=( "$ensureArg" "$ensureVal" )
}
```

The other option is to parametrize _mongod_hack_ensure_arg_val function to process array as a parameter